python/_eos.cc

```diff
@@ -211,9 +211,28 @@ BOOST_PYTHON_MODULE(_eos)
         .def("__getitem__", (Parameter (Parameters::*)(const QualifiedName &) const) &Parameters::operator[])
         .def("by_id", (Parameter (Parameters::*)(const Parameter::Id &) const) &Parameters::operator[])
         .def("__iter__", range(&Parameters::begin, &Parameters::end))
-        .def("declare", &Parameters::declare, return_value_policy<return_by_value>(),
+        .def("declare", &Parameters::declare,
             R"(
-            Declare a new parameter.
+            Declare a new parameter as part of the default parameter set.
+
+            :param name: The name of the parameter to declare.
+            :type name: eos.QualifiedName
+            :param latex: The LaTeX representation for the parameter.
+            :type latex: str
+            :param unit: The unit in which the parameter is expressed.
+            :type unit: eos.Unit
+            :param value: The initial value for the parameter.
+            :type value: float
+            :param min: The minimum value that the parameter can attain.
+            :type min: float
+            :param max: The maximum value that the parameter can attain.
+            :type max: float
```

Comment:
Shouldn't these specifications be optional?

Reply:
Hm, probably. How do you do default parameter values in boost-python?

Reply:
With colons? https://www.boost.org/doc/libs/1_83_0/libs/parameter_python/doc/html/index.html#id10

Reply:
Nope. https://boostorg.github.io/python/doc/html/tutorial/tutorial/functions.html#tutorial.functions.default_arguments.boost_python_member_function_ove , I am not a fan...

Since this is a low-level function that should ideally not be used by users directly, I think we should make callers of ``eos.Parameters.declare_and_insert`` provide all arguments in the Python API.